src/reanimated2/hook/useWorkletCallback.ts

```diff
@@ -2,6 +2,9 @@
 import { useCallback } from 'react';
 import type { DependencyList } from './commonTypes';
 
+/**
+ * @deprecated don't use
```

Comment:
im a bit late for this change, but what is the recommended replacement for this hook ? thanks

Reply:
@gorhom The recommended way is to just use `useCallback` hook with `'worklet';` directive.

`useWorkletCallback` used to do two extra things:
1. the callback is automatically treated as a worklet, similarly to `useAnimatedStyle`
2. the dependency array is filled automatically, also similarly to `useAnimatedStyle`

While the hook is probably safe for basic use, @szydlovsky found out that in some cases `useWorkletCallback` **does not behave correctly**. For instance, when extra items are declared in the dependency array, the dependencies found by Reanimated Babel plugin are not taken into account. First, we tried to eliminate this bug but eventually gave up because of too many cases to consider and validate (web with plugin, web without plugin, native with plugin; no dependency array, empty dependency array, non-empty dependency array etc.).

We don't see any particular value in this hook since it just saves one line of code (i.e. `'worklet;'` directive). With that in mind, we plan to provide an extendable mechanism for library developers to register custom hooks to have their arguments automatically workletized. This would be highly beneficial for RNVisionCamera, RNSkia, RNKeyboardController and other libraries which build on top of worklets without needing to explicitly list such hooks on Reanimated side (which also requires to publish a new release for such changes to take account).

At this point we strongly encourage to use `useWorklet` + `'worklet';` directive in new code as well as migrate old code using `useWorkletCallback` if possible.